test/inductor/test_cpu_repro.py

```diff
@@ -4151,6 +4151,46 @@ def forward(self, x):
         x = torch.randn(1, 4, 2, 2)
         self.common(fn, (x,))
 
+    @parametrize("is_inference", (True, False))
+    def test_disabled_amp(self, is_inference):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.all_head_size = 12 * 64
+                self.dense = nn.Linear(self.all_head_size, self.all_head_size)
+
+            def forward(self, q, k, v):
+                context_layer = F.scaled_dot_product_attention(
+                    q, k, v, attn_mask=None, dropout_p=0.2
+                )
+                context_layer = context_layer.permute(0, 2, 1, 3).contiguous()
+                new_context_layer_shape = context_layer.size()[:-2] + (
+                    self.all_head_size,
+                )
+                context_layer = context_layer.view(new_context_layer_shape)
+                return self.dense(context_layer)
+
+        mod = M().to(torch.bfloat16).eval()
+
+        q = torch.randn((4, 12, 512, 64), dtype=torch.bfloat16) / 10.0
+        k = torch.randn((4, 12, 512, 64), dtype=torch.bfloat16) / 10.0
+        v = torch.randn((4, 12, 512, 64), dtype=torch.bfloat16) / 10.0
+        inputs = (
+            q,
+            k,
+            v,
+        )
+        compiler_mode = torch.compile(mod)
+
+        context = contextlib.nullcontext if not is_inference else torch.no_grad
```

Comment:
I think you should add 

 sdpa_kernel(SDPBackend.MATH):
 
 from torch.nn.attention import sdpa_kernel, SDPBackend
 
 since this only is  an issue for the math.backend and https://github.com/pytorch/pytorch/issues/133974#issuecomment-2305331763 works on 2.5 for for the fused kernels

Reply:
makes sense but trying to get it landed today, dont want to restart tests.

Reply:
nvm, have to rebase anyway